### PR TITLE
Juay hee ugdg update

### PR DESCRIPTION
**Pull request by juayhee**
1. PR related to #70 
2. Updated DG
3. Upcoming: Update UG

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -211,6 +211,38 @@ The steps above are summarised using a sequence diagram as shown below.
...
+Step 1: The user inputs `tag-add 1 t/friend` to add the tag "friend" to the first contact in the list.
+
+Step 2: This argument is passed into `LogicManager` which calls on `Coach2K22Parser#parseCommand()` to find a suitable parser class to process the user inputs. This initialises the `AddTagCommandParser` where its method `#parse` is called to process the user inputs.
+
+Step 3: It then returns a newly initialised `AddTagCommand` back to the `LogicManager` for command execution. This `AddTagCommand` contains information about the new tag (in this case, "friend")
+
+Step 4: During the command execution, the `ModelManager#setPerson()` is called which edits the tags of the person with the user-supplied tags. The filtered person list is updated with `ModelManager#updateFilteredPersonList` to display the new information to the user.
+
+The steps above are summarised using a sequence diagram as shown below.
+![AddTagSequenceDiagram] (images/AddTagSequenceDiagram.png)
```
Comment (Rye-Catcher) [inline docs/DeveloperGuide.md:231]:
```suggestion
![AddTagSequenceDiagram](images/AddTagSequenceDiagram.png)
```

File: docs/UserGuide.md [hunk 2]
```diff
@@ -306,11 +306,29 @@ Examples:
...
+Format: `tag-del-t INDEX [t/TAG]`
+
+Examples:
+* `tag-add-t 1 important` removes the tag "important" from the first task in the list.
+
+<div markdown="block" class="alert alert-info">
+
+**:information_source: Key things to take note of:**<br>
+
+* `TAG must be an exact match for it to be recognised.
```
Comment (Rye-Catcher) [inline docs/UserGuide.md:330]:
```suggestion
* `TAG` must be an exact match for it to be recognised.
```

**Review by Rye-Catcher — COMMENTED** [review]
Overall good. Except for some minor errors. And I think the use cases for `tag-add-t` and `tag-del-t` is not updated in DG?

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (juayhee) [reply docs/DeveloperGuide.md:231]: Fixed

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -211,6 +211,38 @@ The steps above are summarised using a sequence diagram as shown below.
...
 
+### 4.3 Add tags feature
+
+#### 4.3.1 Implementation
+This feature allows the user to add tags to contacts in the list. It is facilitated by `ModelManager` which
+makes use of the method `#setPerson()` and `#updateFilteredPersonList()` to add tags to a contact.
+
+Given below is an example usage scenario of how the add tag mechanism behaves at each step.
+
+Step 1: The user inputs `tag-add 1 t/friend` to add the tag "friend" to the first contact in the list.
```
Comment (simjunheng) [inline docs/DeveloperGuide.md:222]: Should the command format be tag-add-p 1 friend instead?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -604,6 +636,41 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1. User requests to list tasks
+2. Coach2K22 shows a list of tasks
+3. User requests to add a tag to a task of a specified index in the task list
+4. Coach2k22 shows the updated details of the task list
+
+  Use case ends.
+
+**Extensions**
+
+* 3a. The task list is empty.
```
Comment (simjunheng) [inline docs/DeveloperGuide.md:652]: Should this be 2a instead of 3a?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -604,6 +636,41 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+1. User requests to list tasks
+2. Coach2K22 shows a list of tasks
+3. User requests to remove a tag from a task of a specified index in the task list
+4. Coach2K22 shows the updated details of the task list
+
+   Use case ends.
+
+**Extensions**
+
```
Comment (simjunheng) [inline docs/DeveloperGuide.md:668]: Might want to include more extensions for this e.g. invalid index, empty list.

File: docs/UserGuide.md [hunk 6]
```diff
@@ -306,11 +306,29 @@ Examples:
...
+Examples:
+* `tag-add-t 1 important` adds the tag "important" to the first task in the list.
+
 
 #### 2.2.7 Deleting a tag : `tag-del-t`
 
-* (To be filled up by @juayhee)
+Delete a tag from a selected task in our task list.
+
+Format: `tag-del-t INDEX [t/TAG]`
```
Comment (simjunheng) [inline docs/UserGuide.md:321]: Should this be tag-del-t INDEX TAG_NAME instead?

File: docs/diagrams/AddTagSequenceDiagram.puml [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+activate Coach2K22Parser
+
+create AddTagCommandParser
+Coach2K22Parser -> AddTagCommandParser
+activate AddTagCommandParser
+
+AddTagCommandParser --> Coach2K22Parser
+deactivate AddTagCommandParser
+
+Coach2K22Parser -> AddTagCommandParser : parse("1 t/friend")
```
Comment (simjunheng) [inline docs/diagrams/AddTagSequenceDiagram.puml:29]: Might want to update the format here as well

**Review by simjunheng — CHANGES_REQUESTED** [review]
Other than some of the issues that I've pointed out, I think it is ready to be merged. Thanks!

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (juayhee) [reply docs/DeveloperGuide.md:222]: Fixed

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (juayhee) [reply docs/DeveloperGuide.md:652]: Fixed

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (juayhee) [reply docs/DeveloperGuide.md:668]: Added more extensions

File: docs/UserGuide.md (hunk 6 above)
Comment (juayhee) [reply docs/UserGuide.md:321]: Fixed

File: docs/diagrams/AddTagSequenceDiagram.puml (hunk 7 above)
Comment (juayhee) [reply docs/diagrams/AddTagSequenceDiagram.puml:29]: Fixed

Comment (juayhee) [reply docs/diagrams/AddTagSequenceDiagram.puml]: Fixed

File: docs/UserGuide.md [hunk 8]
```diff
@@ -306,11 +306,29 @@ Examples:
 
 #### 2.2.6 Adding a tag : `tag-add-t`
 
-* (To be filled up by @juayhee)
+Add a tag to a selected task from our task list.
+
+Format: `tag-add-t INDEX [t/TAG]`
```
Comment (simjunheng) [inline docs/UserGuide.md:311]: Should this be tag-add-t INDEX TAG_NAME instead?

File: docs/UserGuide.md [hunk 9]
```diff
@@ -447,16 +465,16 @@ Format: `exit`
...
-| **Find**     | `find-t [n/NAME]…​ [t/TAG]…​`<br> e.g., `find-p n/Meeting n/Training t/team1 t/team2`|
+| Action       | Format, Examples                                                                      |
+|--------------|---------------------------------------------------------------------------------------|
+| **List**     | `list-p`                                                                              |
+| **Add**      | (to be filled up by @pragya)                                                          |
+| **Del**      | (to be filled up by @pragya)                                                          | 
+| **Edit**     | (to be filled up by @pragya)                                                          |
+| **Clear**    | `clear-t [d/DATE]`<br> e.g., `clear-t 2022-10-10`                                     |
+| **Add Tag**  | `tag-add-t INDEX [t/TAG]` <br> e.g., `tag-add-t 1 t/important`                        |
+| **Del Tag**  | `tag-del-t INDEX [t/TAG]` <br> e.g., `tag-del-t 1 t/important`                        |
```
Comment (simjunheng) [inline docs/UserGuide.md:476]: Do update the changes to the command format in your command summary as well

File: docs/diagrams/AddTagSequenceDiagram.puml [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+participant ":AddTagCommandParser" as AddTagCommandParser LOGIC_COLOR
+participant "p:AddTagCommand" as AddTagCommand LOGIC_COLOR
+participant ":CommandResult" as CommandResult LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant ":Model" as Model MODEL_COLOR
+end box
+
+[-> LogicManager : execute("tag-add 1 t/friend")
```
Comment (simjunheng) [inline docs/diagrams/AddTagSequenceDiagram.puml:16]: Do update your sequence diagram with the new command format as well

**Review by simjunheng — APPROVED** [review]

**Pull request merged**